subsys/bluetooth/host/scan.h

```diff
@@ -5,12 +5,95 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#ifndef SUBSYS_BLUETOOTH_HOST_SCAN_H_
+#define SUBSYS_BLUETOOTH_HOST_SCAN_H_
+
+#include <stdint.h>
+
+#include <zephyr/sys/atomic.h>
+#include <zephyr/bluetooth/bluetooth.h>
+
+/**
+ * Reasons why a scanner can be running.
+ * Used as input to @ref bt_le_scan_user_add
+ * and @ref bt_le_scan_user_remove.
+ */
+enum bt_le_scan_enabled_flag {
+	/** The application explicitly instructed the stack to scan for advertisers
+	 * using the API @ref bt_le_scan_start().
+	 * May not be used explicitly.
+	 */
+	BT_LE_SCAN_USER_EXPLICIT_SCAN,
+
+	/**
+	 * Syncing to a periodic advertiser.
+	 */
+	BT_LE_SCAN_USER_SYNC_SYNCING,
+
+	/**
+	 * Scanning to find devices to connect to.
+	 */
+	BT_LE_SCAN_USER_SCAN_BEFORE_INITIATE,
+
+	/**
+	 * Special state for a NOP for @ref bt_le_scan_update_and_reconfigure to not add/remove any
+	 * flag why the scanner should be running.
+	 */
+	BT_LE_SCAN_USER_NONE,
```

Comment:
I'm still having some trouble understanding the `NONE`. 

Why is it necessary?
What does it mean to add a user with `NONE`, and why is that allowed?
What does it mean to remove a user with `NONE`, and why is that allowed?

Reply:
I can add a comment to the function